iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs

```diff
@@ -223,7 +223,7 @@ public async Task DeviceNotFoundExceptionReturnsDeviceDisabledStatus()
             // act
             await ((Func<Task>)(() => sut
                 .OpenAsync(CancellationToken.None)))
-                .ExpectedAsync<DeviceNotFoundException>()
+                .ExpectedAsync<IotHubClientException>()
```

Comment:
nit: any way for us to check the status code in here?

Reply:
Good catch, I believe I added status code checks in our E2E tests but forgot here. Will do